packages/multichain-account-service/src/providers/SolAccountProvider.ts

```diff
@@ -14,21 +14,43 @@ import type { Json, JsonRpcRequest } from '@metamask/utils';
 import type { MultichainAccountServiceMessenger } from 'src/types';
 
 import { SnapAccountProvider } from './SnapAccountProvider';
+import { withTimeout } from './utils';
+
+export type SolAccountProviderConfig = {
+  discovery: {
+    timeoutMs: number;
+  };
+};
 
 export class SolAccountProvider extends SnapAccountProvider {
+  static NAME = 'Solana' as const;
+
   static SOLANA_SNAP_ID = 'npm:@metamask/solana-wallet-snap' as SnapId;
 
   readonly #client: KeyringClient;
 
-  constructor(messenger: MultichainAccountServiceMessenger) {
+  readonly #config: SolAccountProviderConfig;
+
+  constructor(
+    messenger: MultichainAccountServiceMessenger,
+    config: SolAccountProviderConfig = {
+      discovery: {
+        // Use a pretty high timeout (12s) for each Solana discovery for now.
```

Comment:
12s seems pretty high for each discovery call, that's 24x over EVM (imagine importing 10+ accounts, it adds up quickly). If there's a legitimate failure/slowdown with the discovery on the solana snap side, keeping the timeout this high can mask that problem. Perhaps, the metrics that we spoke about discovery can actually be useful for this scenario. IIRC we were going to add an errors field, reporting these timeout errors in production can make it easier to diagnose if there's an actual problem in the snap; be it some laggy APIs or the actual snap code. I would opt to keep this closer to the EVM timeout, maybe 1.5-2x max that number.